src/inputBackend/input/DevInput.cpp

```diff
@@ -49,7 +49,14 @@ DevInputBase::DevInputBase(const string& name) :
 
 DevInputBase::~DevInputBase()
 {
-	stop();
+	try
+	{
+	    stop();
+	}
+	catch(const std::exception& err )
+	{
+		LOG(mLog, ERROR) << err.what();
```

Comment:
Indentation is not right.

And what about logic? Will be application in well-defined state if `stop()` failed?

Reply:
> Indentation is not right.
> 
> And what about logic? Will be application in well-defined state if `stop()` failed?

It will be in much more better state than before, the fix is just to avoid the crash. 
Besides, now log has the information what is going on there.
Saying about the logic to improve - the new issue must be created to redesign the algorithm.

Reply:
But this affects the invariant. Your program is no longer in the correct state. You either should ensure that
exception handler will leave the program in the a defined state, or that it will crash it. There is no sense in catching exception and pretending that all is fine.

Reply:
is it possible to extend the issue with your vision ? i will continue the investigation.